enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -622,58 +622,178 @@ will be reconciled in any namespaces, removing the need for ServiceMeshMemberRol
 among Gateway API implementations due to the fact manually deployments require linking an
 Istio-specific service address.
 
+#### Security Policy
+
+##### Allowing All Namespaces for Gateways
+
+Unlike OpenShift ingress, neither Gateway API nor Istio restrict Gateway objects to specific namespaces. This design
+allows for different operational groups to create and manage their own [dedicated Gateway](#dedicated-gateway-topology)
+in an application namespace rather than relying on a [shared Gateway](#shared-gateway-topology) residing in another
+namespace. This departs from the current OpenShift ingress approach of all router pods being confined to the
+`openshift-ingress` namespace. Our design will allow Gateways to be created in any namespace, provided the user has
+sufficient [RBAC](#rbac) permissions.
+
+###### Limiting DNS Across Namespaces
+
+As mentioned in [New Controller to Manage DNS Records for Gateway Listeners](#new-controller-to-manage-dns-records-for-gateway-listeners),
+the Ingress Operator automatically creates DNS records for Gateways. However, for our initial implementation, this
+will be limited to the `openshift-ingress` namespace to avoid introducing complexities for DNS management while
+Gateway API upstream is still defining guidelines for DNS record management (see
+[kubernetes-sigs/gateway-api#2627](https://github.com/kubernetes-sigs/gateway-api/issues/2627)).
+
+###### Gateway Merging Across Namespaces
+
+Allowing Gateways in all namespaces can pose a potential security risk for Gateways that merge listeners
+across namespaces, as one Gateway or listener could preempt another listener in a different namespace. However,
+there currently isn't a way to merge Gateways across namespaces in Istio as the `spec.addresses`
+field is [restricted](https://github.com/istio/istio/blob/915ac2bf05dbc7a08ae61a8bc39fcd6ea1f3d11a/pilot/pkg/config/kube/gateway/context.go#L78)
+to Gateways in the same namespace. Work is underway to develop a mechanism for
+[Gateway merging](https://docs.google.com/document/d/1qj7Xog2t2fWRuzOeTsWkabUaVeOF7_2t_7appe8EXwA/edit?usp=sharing),
+and it is important to review the outcome for any potential security implications.
+
+##### RBAC
+
+OpenShift ships with a set of [default ClusterRoles](https://docs.openshift.com/container-platform/latest/post_installation_configuration/preparing-for-users.html#default-roles_post-install-preparing-for-users)
+to enable users to implement RBAC. Gateway API has its own [RBAC recommendation](https://gateway-api.sigs.k8s.io/concepts/security-model/#rbac)
+defining which personas get write permission for each object. Gateway API users in OpenShift will need the ability to
+implement one of these security models. To support this, the Ingress Operator will add additional RBAC
+permissions for the following default ClusterRoles:
+
+| OpenShift ClusterRole | GatewayClass | Gateway              | xRoute Types         | ReferenceGrant |
```

Comment:
ReferenceGrant is the only way an application developer (or admin) can allow xRoute access to a Service they own/manage in a different namespace.  I think we need to re-consider allowing only cluster-admin access to ReferenceGrants.

From the [docs](https://gateway-api.sigs.k8s.io/api-types/referencegrant/?h=referencegrant): "A ReferenceGrant can be used to enable cross namespace references within Gateway API. In particular, Routes may forward traffic to backends in other namespaces, or Gateways may refer to Secrets in another namespace".  Routes being able to forward traffic to backends in other namespaces may be useful to application developers trying to grant access to their applications.

Reply:
If this becomes a sticking point we should consider pulling ReferenceGrant support out for GA so we can make an iteration specifically about it. We need to ensure that we don't allow cross namespace references by default as a consequence of such a thing, however.

Reply:
No, I don't think it would be necessary to add more work by removing the ReferenceGrant CRD.  Let's discuss the merits of allowing app developers and admins access to ReferenceGrant.   If it's easier to bar it in the beginning and open it up later, let's do that.

Reply:
To be clear, we're talking about whether the _default_ RBAC policy will allow creating referencegrants, right?  That is, the CRD will be there in any case, and the cluster-admin could define RBAC to allow broader use of referencegrants, right?